paasta_tools/docker_hostname_wrapper.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+""" Meant to be used by mesos-slave instead of the /usr/bin/docker executable
+directly This will parse the CLI arguments intended for docker, extract
+environment variable settings related to the actual node hostname and mesos
+task ID, and use those as an additional --hostname argument when calling the
+underlying docker command.
+
+If the environment variables are unspecified, or if --hostname is already
+specified, this does not change any arguments and just directly calls docker
+as-is.
+"""
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import os
+import re
+import socket
+import sys
+
+
+ENV_MATCH_RE = re.compile('^(-\w*e\w*|--env)(=(\S.*))?$')
+MAX_HOSTNAME_LENGTH = 63
+
+
+def parse_env_args(args):
+    result = {}
+    in_env = False
+    for arg in args:
+        if not in_env:
+            match = ENV_MATCH_RE.match(arg)
+            if not match:
+                continue
+            arg = match.group(3) or ''
+            if not arg:
+                in_env = True
+                continue
+
+        in_env = False
+        if '=' not in arg:
+            continue
+
+        k, _, v = arg.partition('=')
+        result[k] = v
+
+    return result
```

Comment:
One idea: instead of rolling your own argument parsing,  argparse + parse_known_args might be a good idea?